users/shared_account/shared_account.go

```diff
@@ -0,0 +1,166 @@
+
+	"github.com/trackit/trackit-server/routes"
+	"github.com/trackit/trackit-server/users"
+	"github.com/trackit/trackit-server/db"
+)
+
+// inviteUserRequest is the expected request body for the invite user route handler.
+type InviteUserRequest struct {
+	Email           string `json:"email" req:"nonzero"`
+	AccountId       int    `json:"accountId"`
+	PermissionLevel int    `json:"permissionLevel"`
+}
+
+type updateUsersSharedAccountRequest struct {
+	ShareId         int `json:"shareId" req:"nonzero"`
+	PermissionLevel int `json:"permissionLevel"`
+}
+
+type deleteUsersSharedAccountRequest struct {
+	ShareId         int `json:"shareId" req:"nonzero"`
+}
+
+func init() {
+	routes.MethodMuxer{
+		http.MethodGet: routes.H(listSharedUsers).With(
+			db.RequestTransaction{db.Db},
+			users.RequireAuthenticatedUser{users.ViewerAsParent},
+			routes.Documentation{
+				Summary:     "List shared users",
+				Description: "Return a list of user who have an access to an AWS account on Trackit",
+			},
+			routes.QueryArgs{
+				routes.AwsAccountIdQueryArg,
+			},
+		),
+		http.MethodPost: routes.H(inviteUser).With(
+			db.RequestTransaction{db.Db},
+			users.RequireAuthenticatedUser{users.ViewerAsParent},
+			routes.RequestContentType{"application/json"},
+			routes.RequestBody{InviteUserRequest{"example@example.com", 1234, 0}},
+			routes.Documentation{
+				Summary:     "Creates an invite",
+				Description: "Creates an invite for account team sharing",
+			},
+		),
+		http.MethodPatch: routes.H(updateSharedUsers).With(
+			db.RequestTransaction{db.Db},
+			users.RequireAuthenticatedUser{users.ViewerAsParent},
+			routes.RequestContentType{"application/json"},
+			routes.RequestBody{updateUsersSharedAccountRequest{1, 2}},
+			routes.Documentation{
+				Summary:     "Update shared users",
+				Description: "Update shared users associated with a specific AWS account",
+			},
+		),
+		http.MethodDelete: routes.H(deleteSharedUsers).With(
+			db.RequestTransaction{db.Db},
+			users.RequireAuthenticatedUser{users.ViewerAsParent},
+			routes.RequestBody{deleteUsersSharedAccountRequest{1}},
+			routes.Documentation{
+				Summary:     "Delete shared users",
+				Description: "Delete shared users associated with a specific AWS account",
+			},
+		),
+	}.H().With(
+		db.RequestTransaction{db.Db},
+		routes.Documentation{
+			Summary: "interact with shared accounts",
+		},
+	).Register("/user/share")
+}
+
+// inviteUser handles users invite for team sharing.
+func inviteUser(request *http.Request, a routes.Arguments) (int, interface{}) {
+	var body InviteUserRequest
+	routes.MustRequestBody(a, &body)
+	tx := a[db.Transaction].(*sql.Tx)
+	user := a[users.AuthenticatedUser].(users.User)
+	return InviteUserWithValidBody(request, body, tx, user)
+}
+
+// listSharedUsers handles listing of users who have an access to an AWS account.
+func listSharedUsers(request *http.Request, a routes.Arguments) (int, interface{}) {
+	body := a[routes.AwsAccountIdQueryArg].(int)
```

Comment:
`body` is not a good name, you should put `awsAccountId` or something like this.